components/sup/src/manager/user_config_watcher.rs

```diff
@@ -76,21 +79,26 @@ struct WorkerState {
 
 type ServiceName = String;
 pub struct UserConfigWatcher {
-    states: HashMap<ServiceName, WorkerState>,
+    // We use Arc/Mutex here, because this needs to be shareable
+    // across threads so we can remove watchers from futures; the
+    // Sender and Receiver members of WorkerState aren't themselves
+    // shareable.
+    states: Arc<Mutex<HashMap<ServiceName, WorkerState>>>,
```

Comment:
Why not a doc comment?

Reply:
Only because it's more of an implementation detail for developers, rather than anything describing the fundamental intention of the data itself.

Reply:
Fair. I don't know if anyone actually reads out documentation compiled from doc comments, but if they did, I figured they might wonder why that type since that's part of the info they'd be looking at.